packages/cardhost/app/components/mode-switcher.js

```diff
@@ -0,0 +1,35 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+const modes = [
+  {
+    id: 'view',
+    displayTitle: 'Layout View'
+  },
+  {
+    id: 'edit',
+    displayTitle: 'Edit View'
+  },
+  {
+    id: 'schema',
+    displayTitle: 'Schema View'
+  }
+];
+
+export default class ModeSwitcher extends Component {
+  modes = modes;
+
+  @service router;
+
+  constructor(...args) {
+    super(...args);
+  }
```

Comment:
no need for this. if you don't provide a constructor, it will automatically do what you have here. you can just delete the constructor.